JSDemos/Demos/Toolbar/Multiline/React/App.js

```diff
@@ -0,0 +1,323 @@
+import SelectBox from 'devextreme-react/select-box';
+import notify from 'devextreme/ui/notify';
+import 'devextreme/ui/select_box';
+import {
+  fontSizes,
+  lineHeights,
+  fontFamilies,
+  fontStyles,
+  textAlignItems,
+  textAlignItemsExtended,
+  listTypes,
+} from './data.js';
+
+const lineHeightDefault = lineHeights[1].lineHeight;
+const textAlignDefault = [textAlignItems[0].alignment];
+const fontSizeDefault = fontSizes[2].size;
+
+const toolbarLineModes = [
+  {
+    text: 'Multiline mode',
+    value: true,
+  },
+  {
+    text: 'Singleline mode',
+    value: false,
+  },
+];
+
+const attachButtonOptions = {
+  icon: 'attach',
+  text: 'Attach',
+  onClick: () => notify('Attach button has been clicked!'),
+};
+
+const addButtonOptions = {
+  icon: 'add',
+  text: 'Add',
+  onClick: () => notify('Add button has been clicked!'),
+};
+
+const removeButtonOptions = {
+  icon: 'trash',
+  text: 'Remove',
+  onClick: () => notify('Remove button has been clicked!'),
+};
+
+const aboutButtonOptions = {
+  icon: 'help',
+  text: 'About',
+  onClick: () => notify('About button has been clicked!'),
+};
+
+function App() {
+  const [lineHeight, setLineHeight] = React.useState(lineHeightDefault);
+  const [textAlign, setTextAlign] = React.useState(textAlignDefault);
+  const [fontSize, setFontSize] = React.useState(fontSizeDefault);
+  const [multiline, setMultiline] = React.useState(true);
+  const [checkBoxValue, setCheckBoxValue] = React.useState(false);
+
+  const onItemClick = React.useCallback((e) => {
+    notify(`The "${e.itemData.hint}" button was clicked`);
+  }, []);
+
+  const onUndoButtonClick = React.useCallback(() => {
+    notify('Undo button has been clicked!');
+  }, []);
+
+  const onRedoButtonClick = React.useCallback(() => {
+    notify('Redo button has been clicked!');
+  }, []);
+
+  const onCheckBoxChange = React.useCallback(({ value }) => {
+    setCheckBoxValue(value);
+    notify('Checkbox value has been changed!');
+  }, []);
+
+  const onLineHeightChange = React.useCallback((e) => {
+    setLineHeight(e.item.lineHeight);
+    notify('Line height value has been changed!');
+  }, []);
+
+  const onFontSizeChange = React.useCallback((e) => {
+    setFontSize(e.item.size);
+    notify('Font size value has been changed!');
+  }, []);
+
+  const onTextAlignChange = React.useCallback((e) => {
+    setTextAlign([e.itemData.alignment]);
+  }, []);
+
+  const onToolbarLineModeChange = React.useCallback(({ value }) => {
+    setMultiline(value);
+  }, []);
+
+  const renderFontSize = React.useCallback((itemData) => (
+    <div style={{ fontSize: `${itemData.size}px` }}>{itemData.text}</div>
+  ), []);
+
+  const renderTextAlign = React.useCallback(() => (
+    <ButtonGroup
+      keyExpr="alignment"
+      stylingMode="outlined"
+      items={textAlignItems}
+      selectedItemKeys={textAlign}
+      onItemClick={onTextAlignChange}
+    ></ButtonGroup>
+  ), [textAlign]);
```

Comment:
As far as I know, it's good practice to add 'onTextAlignChange' to 'useCallback' dependencies too.